src/core/mantle/data/csv.test.ts

```diff
@@ -0,0 +1,310 @@
+    },
+    {
+      id: "3",
+      geometry: {
+        type: "Point",
+        coordinates: [4, undefined],
+      },
+      properties: {
+        country: "UK",
+        lng: "5",
+      },
+      range: undefined,
+    },
+  ]);
+});
+
+test("has multiline field", async () => {
+  const fetchDataMock = vi.spyOn(Utils, "f");
+  fetchDataMock.mockImplementation(async () => {
+    return {
+      // lat has no header name
+      text: async () => `id,country,lat,text,lng
+1,Japan,0,"Hello
+World",1
+2,US,2,"This is list
+- a
+- b
+- c
+",3
+3,UK,4,"Hey,
+I'm CSV",5
+`,
+    } as Response;
+  });
+
+  const features = await fetchCSV({
+    type: "csv",
+    url: "http://example.com",
+    csv: {
+      idColumn: 0,
+      latColumn: 2,
+      lngColumn: 4, // This should not fount.
+      noHeader: false,
+    },
+  });
+
+  expect(features).toEqual([
+    {
+      id: "1",
+      geometry: {
+        type: "Point",
+        coordinates: [0, 1],
+      },
+      properties: {
+        country: "Japan",
+        text: `Hello
+World`,
+      },
+      range: undefined,
+    },
+    {
+      id: "2",
+      geometry: {
+        type: "Point",
+        coordinates: [2, 3],
+      },
+      properties: {
+        country: "US",
+        text: `This is list
+- a
+- b
+- c
+`,
+      },
+      range: undefined,
+    },
+    {
+      id: "3",
+      geometry: {
+        type: "Point",
+        coordinates: [4, 5],
+      },
+      properties: {
+        country: "UK",
+        text: `Hey,
+I'm CSV`,
+      },
+      range: undefined,
+    },
+  ]);
+});
+
+test("without header", async () => {
+  const fetchDataMock = vi.spyOn(Utils, "f");
+  fetchDataMock.mockImplementation(async () => {
+    return {
+      text: async () => `1,Japan,0,,1
+2,US,2,,3
+3,UK,4,,5
+`,
+    } as Response;
+  });
+
+  const features = await fetchCSV({
+    type: "csv",
+    url: "http://example.com",
+    csv: {
+      idColumn: 0,
+      latColumn: 2,
+      lngColumn: 4,
+      noHeader: true,
+    },
+  });
+
+  expect(features).toEqual([
+    {
+      id: "1",
+      geometry: {
+        type: "Point",
+        coordinates: [0, 1],
+      },
+      properties: {},
+      range: undefined,
+    },
+    {
+      id: "2",
+      geometry: {
+        type: "Point",
+        coordinates: [2, 3],
+      },
+      properties: {},
+      range: undefined,
+    },
+    {
+      id: "3",
+      geometry: {
+        type: "Point",
+        coordinates: [4, 5],
+      },
+      properties: {},
+      range: undefined,
+    },
+  ]);
+});
+
+test("some delimiter", async () => {
+  const fetchDataMock = vi.spyOn(Utils, "f");
+  fetchDataMock.mockImplementation(async () => {
+    return {
+      text: async () => `1,Japan,0,,1
+2;US;2;;3
+3\tUK\t4\t\t5
+`,
+    } as Response;
+  });
+
+  const features = await fetchCSV({
+    type: "csv",
+    url: "http://example.com",
+    csv: {
+      idColumn: 0,
+      latColumn: 2,
+      lngColumn: 4,
+      noHeader: true,
```

Comment:
Note: If idColumn is not specified, id should be automatically generated (uuid v4)

Reply:
I've added the test case in here.
https://github.com/reearth/reearth-web/pull/382/commits/67618d21d65ab6ae95a4ed99824e2ef14147ffd4#diff-23b72191b57013c2f52595d1869fb1b02ba09ab2d71af89f93b8f21bf1bde3f7R316